src/main/java/com/google/sps/servlets/NewClass.java

```diff
@@ -44,18 +49,46 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     try {
       String className = request.getParameter("className").trim();
 
-      String idToken = request.getParameter("idToken");
-      FirebaseToken decodedToken = authInstance.verifyIdToken(idToken);
+      // prevents creating duplicate classes
+      Query query =
+          new Query("Class")
+              .setFilter(new FilterPredicate("name", FilterOperator.EQUAL, className));
+      Boolean canAdd = true;
+      if (datastore.prepare(query).countEntities() != 0) {
+        canAdd = false;
+      }
 
-      Entity classEntity = new Entity("Class");
-      classEntity.setProperty("owner", decodedToken.getUid());
-      classEntity.setProperty("name", className);
-      classEntity.setProperty("beingHelped", "");
-      classEntity.setProperty("studentQueue", Collections.emptyList());
+      if (canAdd) {
```

Comment:
You can get rid of `canAdd` and inline `datastore.prepare(query).countEntities() == 0` here